File: app/src/main/java/com/mundi/databaseadmin/bean/CellData.java
package com.mundi.databaseadmin.bean;

public class CellData {
    private String datastr;
    private int row;
    private int column;

    public CellData(int row, int column, String datastr) {
        this.row = row;
        this.column = column;
        this.datastr = datastr;
    }

    public CellData(int row, int column, short data) {
        this.row = row;
        this.column = column;
        this.datastr = Short.toString(data);
    }

    public CellData(int row, int column, int data) {
        this.row = row;
        this.column = column;
        this.datastr = Integer.toString(data);
    }

    public CellData(int row, int column, long data) {
        this.row = row;
        this.column = column;
        this.datastr = Long.toString(data);
    }

    public CellData(int row, int column, float data) {
        this.row = row;
        this.column = column;
        this.datastr = Float.toString(data);
    }

    public CellData(int row, int column, double data) {
        this.row = row;
        this.column = column;
        this.datastr = Double.toString(data);
    }

    public CellData(int row, int column, boolean data) {
        this.row = row;
        this.column = column;
        if (data) this.datastr = "true";
        else this.datastr = "false";
    }

    public int getRow() {
        return row;
    }

    public int getColumn() {
        return column;
    }

    public String getData() {
        return datastr;
    }

    public short getDataShort() {
        return Short.parseShort(datastr);
    }

    public int getDataInteger() {
        return Integer.parseInt(datastr);
    }

    public long getDataLong() {
        return Long.parseLong(datastr);
    }

    public float getDataFloat() {
        return Float.parseFloat(datastr);
    }

    public double getDataDouble() {
        return Double.parseDouble(datastr);
    }

    public boolean getDataBoolean() {
        if (datastr.equalsIgnoreCase("true")) {
            return true;
        } else {
            return false;
        }
    }
}

File: app-kotlin/src/main/java/com/mundi/databaseadmin/bean/CellData.kt
package com.mundi.databaseadmin.bean

class CellData {
    var data: String? = null
        private set
    var row: Int
        private set
    var column: Int
        private set

    constructor(row: Int, column: Int, datastr: String?) {
        this.row = row
        this.column = column
        data = datastr
    }

    constructor(row: Int, column: Int, data: Short) {
        this.row = row
        this.column = column
        this.data = java.lang.Short.toString(data)
    }

    constructor(row: Int, column: Int, data: Int) {
        this.row = row
        this.column = column
        this.data = Integer.toString(data)
    }

    constructor(row: Int, column: Int, data: Long) {
        this.row = row
        this.column = column
        this.data = java.lang.Long.toString(data)
    }

    constructor(row: Int, column: Int, data: Float) {
        this.row = row
        this.column = column
        this.data = java.lang.Float.toString(data)
    }

    constructor(row: Int, column: Int, data: Double) {
        this.row = row
        this.column = column
        this.data = java.lang.Double.toString(data)
    }

    constructor(row: Int, column: Int, data: Boolean) {
        this.row = row
        this.column = column
        if (data) this.data = "true" else this.data = "false"
    }

    val dataShort: Short
        get() = data!!.toShort()

    val dataInteger: Int
        get() = data!!.toInt()

    val dataLong: Long
        get() = data!!.toLong()

    val dataFloat: Float
        get() = data!!.toFloat()

    val dataDouble: Double
        get() = data!!.toDouble()

    val dataBoolean: Boolean
        get() = if (data.equals("true", ignoreCase = true)) {
            true
        } else {
            false
        }
}